sdks/java/core/src/main/java/org/apache/beam/sdk/Pipeline.java

```diff
@@ -503,15 +522,29 @@ void applyReplacement(
     }
   }
 
-  /**
-   * Returns the configured {@link PipelineOptions}.
-   *
-   * @deprecated see BEAM-818 Remove Pipeline.getPipelineOptions. Configuration should be explicitly
-   *     provided to a transform if it is required.
-   */
-  @Deprecated
-  public PipelineOptions getOptions() {
-    return options;
+  private void validate(PipelineOptions options) {
+    for (String uniqueName : unstableNames) {
```

Comment:
This should probably be guarded by the switch, and the error should output a collection of unstable names.